docs/protocol/extension_dynamic_logging.md

```diff
@@ -0,0 +1,177 @@
+settings below. By default "log_file" is unspecified -- "".
+
+- "log_info" : this parameter controls whether the Triton server outputs info messages or
+writes them to the "log_file" (if specified). A value of "false" will stop the Triton server
+from outputting info messages. By default "log_info" is "true".
+
+- "log_warnings" : this parameter controls whether the Triton server outputs warning messages or
+writes them to the "log_file" (if specified). A value of "false" will stop the Triton server
+from outputting warning messages. By default "log_warnings" is "true".
+
+- "log_errors" : this parameter controls whether the Triton server outputs error messages or
+writes them to the "log_file" (if specified). A value of "false" will stop the Triton server
+from outputting error messages. By default "log_errors" is "true".
+
+- "log_verbose_level" : this parameter controls whether the Triton server outputs verbose messages
+of varying degress. This value can be any integer >= 0. If "log_verbose_level" is 0, no verbose messages
+will be output by the Triton server. If "log_verbose_level" is 1, level 1 verbose messages will be output
+by the Triton server. If "log_verbose_level" is 2, the Triton server will output all verbose messages of 
+level <= 2, etc. By default "log_verbose_level" is 0.
+
+- "log_format" : this parameter controls the format of Triton server log messages. There are currently
+2 formats: "default" and "ISO8601".
+
+
+### Log Setting Response JSON Error Object
+
+A failed log setting request will be indicated by an HTTP error status
+(typically 400). The HTTP body must contain the
+$log_setting_error_response object.
+
+```
+$log_setting_error_response =
+{
+  "error": $string
+}
+```
+
+- “error” : The descriptive message for the error.
+
+#### Log Setting Request JSON Object
+
+A log setting request is made with a HTTP POST to
+the dynamic logging endpoint. In the corresponding response, the HTTP body contains the
+response JSON. A successful request is indicated by a 200 HTTP status code.
+
+The request object, identified as $log_setting_request must be provided in the HTTP
+body.
+
+```
+$log_setting_request =
+{
+  $log_setting, ...
+}
+```
+
+When a $log_setting JSON is received (defined above), only the specified
+settings will be updated.
+
+## GRPC
+
+For the trace extension Triton implements the following API:
+
+```
+service GRPCInferenceService
+{
+  …
+
+  // Update and get the trace setting of the Triton server.
+  rpc LogSettings(LogSettingsRequest)
+          returns (LogSettingsResponse) {}
+}
+```
+
+The Log Setting API returns the latest log settings. Errors are indicated
+by the google.rpc.Status returned for the request. The OK code
+indicates success and other codes indicate failure. The request and
+response messages for Log Settings are:
+
+```
+message LogSettingsRequest
+{
+  // The values to be associated with a log setting.
+  message SettingValue
+  {
+    repeated string value = 1;
```

Comment:
Don't need `repeated`, and you should use `oneof` so we can pass in different types (string, bool, etc.)